iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/tapchanger/AbstractTapPositionModification.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification.tapchanger;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.AbstractNetworkModification;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * @author Nicolas PIERRE <nicolas.pierre at artelys.com>
+ */
+public abstract class AbstractTapPositionModification extends AbstractNetworkModification {
+    public static final String TRANSFORMER_STR = "Transformer '";
+    private final String transformerId;
+    private final int tapPosition;
+    /**
+     * Defines the leg on which to apply the change for three winding tranformers. 0, 1 or 2
+     *
+     * @implNote Must NOT be empty if element == TransformerElement.THREE_WINDING_TRANSFORMER
+     */
+    private final ThreeWindingsTransformer.Side legSide;
+
+    /**
+     * @param tapPosition the new tap position
+     * @param legSide     defines on which leg of the three winding transformer the modification will be done.
+     *                    If <code>null</code> on three windings transformer, {@link AbstractTapPositionModification#apply(Network)} will search for a unique rtc.
+     *                    Ignored on two windings transformers.
+     */
+    protected AbstractTapPositionModification(String transformerId, int tapPosition,
+                                              ThreeWindingsTransformer.Side legSide) {
+        this.transformerId = Objects.requireNonNull(transformerId);
+        this.tapPosition = tapPosition;
+        this.legSide = legSide;
+    }
+
+    abstract void applyTwoWindings(Network network, TwoWindingsTransformer twoWindingsTransformer,
```

Comment:
applyTwoWindings to applyTwoWindingsTranformer and applyThreeWindings to applyThreeWindingsTransformer could be better.